snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/tracker/Tracker.java

```diff
@@ -914,11 +914,14 @@ public void setEmitter(@NonNull Emitter emitter) {
 
     /**
      * Whether the session context should be sent with events
-     * @param shouldSend
+     * @param sessionContext
      */
-    public void setSessionContext(boolean shouldSend) {
-        sessionContext = shouldSend;
-        if (trackerSession == null && shouldSend) {
+    public synchronized void setSessionContext(boolean sessionContext) {
```

Comment:
This is a legacy method of the original Tracker class so I wouldn't change the method name as it will be removed, hopefully in the version 3.
However, also the TrackerConfiguration class has the same method signature but we can't change it because Kotlin practices an automatic conversion: https://kotlinlang.org/docs/java-interop.html#getters-and-setters
> Boolean accessor methods (where the name of the getter starts with is and the name of the setter starts with set) are represented as properties which have the same name as the getter method.